internal/util/importutil/import_util.go

```diff
@@ -386,20 +403,31 @@ func GetFileNameAndExt(filePath string) (string, string) {
 	return fileNameWithoutExt, fileType
 }
 
-// getFieldDimension gets dimension of vecotor field
-func getFieldDimension(schema *schemapb.FieldSchema) (int, error) {
+// getFieldParam gets param of field, such as dimension of vecotor field and varchar max length
+func getFieldParam(schema *schemapb.FieldSchema, keyName string) (int, error) {
 	for _, kvPair := range schema.GetTypeParams() {
 		key, value := kvPair.GetKey(), kvPair.GetValue()
-		if key == common.DimKey {
+		if key == keyName {
 			dim, err := strconv.Atoi(value)
 			if err != nil {
-				return 0, fmt.Errorf("illegal vector dimension '%s' for field '%s', error: %w", value, schema.GetName(), err)
+				return 0, fmt.Errorf("illegal value '%s' for field '%s' param '%s', error: %w", value, schema.GetName(), keyName, err)
 			}
 			return dim, nil
 		}
 	}
 
-	return 0, fmt.Errorf("vector dimension is not defined for field '%s'", schema.GetName())
+	return 0, fmt.Errorf("param '%s' is not defined for field '%s'", keyName, schema.GetName())
+}
+
+func getVectorDimension(schema *schemapb.FieldSchema) (int, error) {
+	return getFieldParam(schema, common.DimKey)
+}
+
+func getVarcharMaxLen(schema *schemapb.FieldSchema) (int, error) {
```

Comment:
I recommend adding a function called "getJSONMaxLength" to differentiate it from varchar.